src/rest_vol.c

```diff
@@ -2404,6 +2416,61 @@ RV_find_object_by_path(RV_object_t *parent_obj, const char *obj_path, H5I_type_t
     return ret_value;
 } /* end RV_find_object_by_path() */
 
+/*-------------------------------------------------------------------------
+ * Function:    RV_copy_creation_properties_callback
+ *
+ * Purpose:     Callback for RV_parse_response to get the
+ *              creationProperties field in the response.
+ *
+ *              Allocates memory at *callback_data_out that must be
+ *              freed by caller.
+ *
+ * Return:      Non-negative on success/Negative on failure
+ *
+ * Programmer:  Matthew Larson
+ *              May, 2023
+ */
+herr_t
+RV_copy_creation_properties_callback(char *HTTP_response, void *callback_data_in, void *callback_data_out)
+{
+    yajl_val parse_tree = NULL, key_obj;
+    char    *parsed_string;
+    char   **buf_out   = (char **)callback_data_out;
+    herr_t   ret_value = SUCCEED;
+
+#ifdef RV_CONNECTOR_DEBUG
+    printf("-> Retrieving object's creation properties from server's HTTP response\n\n");
+#endif
+
+    if (!HTTP_response)
+        FUNC_GOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "HTTP response buffer was NULL");
+    if (!buf_out)
+        FUNC_GOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "output buffer was NULL");
+
+    if (NULL == (parse_tree = yajl_tree_parse(HTTP_response, NULL, 0)))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "parsing JSON failed");
+
+    key_obj = yajl_tree_get(parse_tree, object_creation_properties_keys, yajl_t_string);
+
+    if (!YAJL_IS_STRING(key_obj))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_BADVALUE, FAIL, "returned creationProperties is not a string");
+
+    if (NULL == (parsed_string = YAJL_GET_STRING(key_obj)))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_BADVALUE, FAIL, "creationProperties was NULL");
+
+    if (NULL == (*buf_out = RV_malloc(strlen(parsed_string) + 1)))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_CANTALLOC, FAIL, "failed to allocate memory for creationProperties");
+
+    if (NULL == (memcpy(*buf_out, parsed_string, strlen(parsed_string) + 1)))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_SYSERRSTR, FAIL, "failed to copy creationProperties");
+
+done:
```

Comment:
memcpy won't fail and so won't cause an issue here, but it would be good to check for failure so that `*buf_out` can be freed and set to NULL if something fails past allocating it.

Reply:
In general only test code uses the `error:` label. In library code the `done:` label is used as cleanup and so checking for `ret_value < 0` here would be the main way to tell if something failed.